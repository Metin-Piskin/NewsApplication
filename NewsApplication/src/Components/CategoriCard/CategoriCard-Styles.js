import { StyleSheet } from "react-native";

export default StyleSheet.create({
    btnNormalContainer: {
        marginHorizontal: 10,
        marginVertical: 10,
        borderWidth: 1,
        paddingHorizontal: 5,
        paddingVertical: 3,
        borderRadius:5
    },
    btnPressContainer: {
        marginHorizontal: 10,
        marginVertical: 10,
        borderWidth: 1,
        borderColor: 'red',
        paddingHorizontal: 5,
        paddingVertical: 3,
        borderRadius:5
    },
    btnNormal: {
        color: '#000',
        fontSize:15,
        fontFamily: 'CourierPrime-Bold'
    },
    btnPress: {
        color: 'red',
        fontSize:15,
        fontFamily: 'CourierPrime-Bold'
    }
})