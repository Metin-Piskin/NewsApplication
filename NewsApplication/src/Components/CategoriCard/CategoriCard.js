import React from 'react';
import { Text, TouchableOpacity } from 'react-native';

import styles from './CategoriCard-Styles';

const CategoriCard = ({ onPress, label, categori, value }) => {
    return (
        <TouchableOpacity
            onPress={onPress}
            testID='categori-card-press'
            style={[styles.btnNormalContainer, categori === value && styles.btnPressContainer]}
        >
            <Text
                testID='categori-card-label'
                style={[styles.btnNormal, categori === value && styles.btnPress]}
            >
                {label}
            </Text>
        </TouchableOpacity>
    )
}

export default CategoriCard;