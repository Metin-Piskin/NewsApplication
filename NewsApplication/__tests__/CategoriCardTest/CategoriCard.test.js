import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

import CategoriCard from '../../src/Components/CategoriCard';
import CategoriCardStyles from '../../src/Components/CategoriCard/CategoriCard-Styles';

test('renders correctly', () => {
    const comp = render(<CategoriCard />);
    expect(comp).toMatchSnapshot();
});

test('should render press correctly', () => {
    const mockFunction = jest.fn();
    const comp = render(<CategoriCard onPress={mockFunction} />);

    const CategoriCardPress = comp.getByTestId('categori-card-press');
    fireEvent(CategoriCardPress, 'press');

    expect(mockFunction).toBeCalled();
});

test('should render default theme style', () => {
    const test = 'test'
    const comp = render(<CategoriCard categori={test} value={''} />);
    const CategoriCardNormalContainerStyle = comp.getByTestId('categori-card-press').props.style;

    expect(CategoriCardNormalContainerStyle).toMatchObject(CategoriCardStyles.btnNormalContainer);
});

test('should render default theme style', () => {
    const test = 'test'
    const comp = render(<CategoriCard categori={test} value={test} />);
    const CategoriCardPressContainerStyle = comp.getByTestId('categori-card-press').props.style;

    expect(CategoriCardPressContainerStyle).toMatchObject(CategoriCardStyles.btnPressContainer);
});

test('should render default theme style', () => {
    const test = 'test'
    const comp = render(<CategoriCard label={test} />);
    const CategoriCardLabel = comp.getByTestId('categori-card-label').children[0];

    expect(CategoriCardLabel).toBe(test);
});