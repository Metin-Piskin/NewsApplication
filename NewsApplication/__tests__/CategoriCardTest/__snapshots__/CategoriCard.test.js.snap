// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderColor": "red",
      "borderRadius": 5,
      "borderWidth": 1,
      "marginHorizontal": 10,
      "marginVertical": 10,
      "opacity": 1,
      "paddingHorizontal": 5,
      "paddingVertical": 3,
    }
  }
  testID="categori-card-press"
>
  <Text
    style={
      [
        {
          "color": "#000",
          "fontFamily": "CourierPrime-Bold",
          "fontSize": 15,
        },
        {
          "color": "red",
          "fontFamily": "CourierPrime-Bold",
          "fontSize": 15,
        },
      ]
    }
    testID="categori-card-label"
  />
</View>
`;
